include/fastdds/dds/subscriber/DataReader.hpp

```diff
@@ -710,20 +726,91 @@ class DataReader : public DomainEntity
     RTPS_DllAPI ReturnCode_t get_liveliness_changed_status(
             LivelinessChangedStatus& status) const;
 
-    /* TODO
-       RTPS_DllAPI bool get_requested_incompatible_qos_status(
-            fastrtps::RequestedIncompatibleQosStatus& status) const;
+
+    /**
+     * @brief Get the SAMPLE_LOST communication status
+     *
+     * @param [out] status SampleLostStatus object where the status is returned.
+     *
+     * @return RETCODE_OK
      */
+    RTPS_DllAPI ReturnCode_t get_sample_lost_status(
+            SampleLostStatus& status) const;
 
-    /* TODO
-       RTPS_DllAPI bool get_sample_lost_status(
-            fastrtps::SampleLostStatus& status) const;
+    /**
+     * @brief Get the SAMPLE_REJECTED communication status
+     *
+     * @param [out] status SampleRejectedStatus object where the status is returned.
+     *
+     * @return RETCODE_OK
      */
+    RTPS_DllAPI ReturnCode_t get_sample_rejected_status(
+            SampleRejectedStatus& status) const;
 
-    /* TODO
-       RTPS_DllAPI bool get_sample_rejected_status(
-            fastrtps::SampleRejectedStatus& status) const;
+    /**
+     * @brief Returns the subscription matched status
+     * @param[out] status subscription matched status struct
+     * @return RETCODE_OK
+     */
+    RTPS_DllAPI ReturnCode_t get_subscription_matched_status(
+            SubscriptionMatchedStatus& status) const;
+
+    /**
+     * @brief Retrieves in a publication associated with the DataWriter
+     * @param[out] publication_data publication data struct
+     * @param publication_handle InstanceHandle_t of the publication
+     * @return RETCODE_OK
+     *
+     */
+    RTPS_DllAPI ReturnCode_t get_matched_publication_data(
+            PublicationBuiltinTopicData& publication_data,
+            fastrtps::rtps::InstanceHandle_t publication_handle) const;
+
+    /**
+     * @brief Fills the given vector with the InstanceHandle_t of matched DataReaders
+     * @param[out] publication_handles Vector where the InstanceHandle_t are returned
+     * @return RETCODE_OK
+     */
+    RTPS_DllAPI ReturnCode_t get_matched_publications(
+            std::vector<fastrtps::rtps::InstanceHandle_t*>& publication_handles) const;
+
+    /**
+     * @brief This operation creates a ReadCondition. The returned ReadCondition will be attached and belong to the
+     * DataReader.
+     * @param sample_states Vector of SampleStateKind
+     * @param view_states Vector of ViewStateKind
+     * @param instance_states Vector of InstanceStateKind
+     * @return ReadCondition pointer
+     */
+    RTPS_DllAPI ReadCondition* create_readcondition(
+            std::vector<SampleStateKind> sample_states,
+            std::vector<ViewStateKind> view_states,
+            std::vector<InstanceStateKind> instance_states);
```

Comment:
```suggestion
    RTPS_DllAPI ReadCondition* create_readcondition(
            const std::vector<SampleStateKind>& sample_states,
            const std::vector<ViewStateKind>& view_states,
            const std::vector<InstanceStateKind>& instance_states);
```
In general, avoid copy arguments, especially vectors